src/simulator_core/entities/assets/demand_cluster.py

```diff
@@ -29,50 +54,159 @@ def __init__(
+    def create(self) -> None:
+        """Create a representation of the asset in pandapipes.
 
-        :param dict data:dictionary containing the data to be added the asset. The key is the name
-                        of the property, the value of the dict is the value of the property.
-        :return:
-        """
-        pass
+        The DemandCluster asset contains multiple pandapipes components.
 
-    def set_setpoints(self, setpoints: Dict, **kwargs) -> None:
+        The component model contains the following components:
+        - A flow control valve to set the mass flow rate of the system.
+        - A heat exchanger to set the power consumption of the demand
+        - An intermediate junction to link both components.
+        :param pandapipesNet pandapipes_net: pandapipes network object
+        """
+        if not self._initialized:
+            # Create intermediate junction
+            self._intermediate_junction = Junction(
+                pandapipes_net=self.pandapipes_net,
+                pn_bar=self.pressure_input,
+                tfluid_k=self.temperature_supply,
+                height_m=DEFAULT_NODE_HEIGHT,
+                name=f"intermediate_junction_{self.name}",
+            )
+            # Create the control valve
+            self._flow_control = ControlValve(
+                pandapipes_net=self.pandapipes_net,
+                controlled_mdot_kg_per_s=self.mass_flowrate,
+                diameter_m=self.internal_diameter,
+                control_active=True,
+                in_service=True,
+                name=f"flow_control_{self.name}",
+            )
+            self._flow_control.from_junction = self._from_junction
+            self._flow_control.to_junction = self._intermediate_junction
+            self._flow_control.create()
+
+            # Create the heat exchanger
+            self._heat_exchanger = HeatExchanger(
+                pandapipes_net=self.pandapipes_net,
+                diameter_m=self.internal_diameter,
+                heat_flux_w=self.thermal_power_allocation,
+                name=f"heat_exchanger_{self.name}",
+                in_service=True,
+            )
+            self._heat_exchanger.from_junction = self._intermediate_junction
+            self._heat_exchanger.to_junction = self._to_junction
+            self._heat_exchanger.create()
+
+            self._initialized = True
+
+    def set_setpoints(self, setpoints: Dict) -> None:
         """Placeholder to set the setpoints of an asset prior to a simulation.
 
         :param Dict setpoints: The setpoints that should be set for the asset.
             The keys of the dictionary are the names of the setpoints and the values are the values
         """
-        pass
-
-    def get_setpoints(self, **kwargs) -> Dict:
+        self.thermal_power_allocation = setpoints[PROPERTY_HEAT_DEMAND]
+        self.temperature_return_target = setpoints[PROPERTY_TEMPERATURE_RETURN]
```

Comment:
In your current implementation I do not see the need for a temperature_return_target. 

Can you implement it as self.temperature_return? Or do you have an additional reason to have it in this code?

Reply:
i need that when controling pump of demand, i need to achieve desired return temperature

Reply:
Probably you both mean the same thing the desired value for the return temp and then I agree with @octavianor to at target to the name makes it more clear